packages/solana/src/connectors/universalProvider.ts

```diff
@@ -48,4 +36,15 @@ export class UniversalProviderFactory {
       SolStoreUtil.setAddress('')
     })
   }
+
+  public static async getProvider() {
+    if (!UniversalProviderFactory.provider) {
```

Comment:
I think we could wrap the condition in one condition at a level which would make it easier to read



```suggestion
public static async getProvider() {
  if (!UniversalProviderFactory.provider) {
    const provider = await UniversalProviderFactory.init()

    if (!provider) {
      throw new Error('Failed to initialize universal provider')
    }

    return provider
  }

  return UniversalProviderFactory.provider
}
```